site/public/js/websocket.js

```diff
@@ -35,28 +35,33 @@ class WebSocketClient {
         my_url.port = window.websocketPort;
         my_url.pathname = 'ws';
         this.url = my_url.href;
-        this.serverError = false;
+        this.serverError = null;
     }
 
-    open(page) {
+    open(page, args = {}) {
         console.log(`WebSocket: connecting to ${this.url}`);
-        this.client = new WebSocket(this.url);
+        const [term, course] = document.body.dataset.courseUrl.split('/').slice(4);
```

Comment:
Can you think of any better way to get the term/course? This could be a bit unstable if any route structure were to change.